src/listen.js

```diff
@@ -121,21 +121,27 @@ module.exports = function(defaultFuncs, api, ctx) {
               break;
             case 'chatproxy-presence':
               // TODO: what happens when you're logged in as a page?
-              if(!ctx.globalOptions.updatePresence) {
-                return;
-              }
-
-              if (ctx.loggedIn) {
-                for(var userID in v.buddyList) {
-                  var formattedPresence = utils.formatProxyPresence(v.buddyList[userID], userID);
-                  if(formattedPresence != null)
-                  {
-                    globalCallback(null, formattedPresence);
-                  }
-                }
-                return;
-              }
+              case 'buddylist':
```

Comment:
Not sure what's happening here, I think maybe you misunderstood me before.
You should be able to simply replace
```
if(!ctx.globalOptions.updatePresence) {
```

with something like 

```
if(!ctx.globalOptions.updatePresence || ctx.globalOptions.updateBuddyList) {
```